drone-impl/src/main/java/org/jboss/arquillian/drone/impl/DeploymentDronePointsRegistry.java

```diff
@@ -0,0 +1,67 @@
+package org.jboss.arquillian.drone.impl;
+
+import java.lang.annotation.Annotation;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.drone.spi.DronePoint;
+
+/**
+ * This registry keeps all {@link DronePoint}s with the {@link DronePoint.Lifecycle.DEPLOYMENT} scope and the reference
+ * to the instance of the test class the DronePoint is declared in
+ *
+ * @author <a href="mailto:mjobanek@redhat.com">Matous Jobanek</a>
+ */
+public class DeploymentDronePointsRegistry {
```

Comment:
Is there a separate registry per Drone point lifecycle? If not, I'd rather think of having just on registry class.